src/tasks/FilterFeedTask.ts

```diff
@@ -0,0 +1,83 @@
+import { MentionRegistry } from '../mentions/MentionRegistry';
+import { FilterFeedConfig } from '../BotConfig';
+import { Client, TextChannel } from 'discord.js';
+import * as log4js from 'log4js';
+import Task from './Task';
+import JiraClient = require( 'jira-connector' );
+
+export default class FilterFeedTask extends Task {
+	public static logger = log4js.getLogger( 'FilterFeed' );
+
+	private client: Client;
+	private jira: JiraClient;
+	private channel: string;
+	private jql: string;
+	private title: string;
+
+	private knownTickets: Set<string>;
+
+	constructor( { channel, jql, title }: FilterFeedConfig, client: Client ) {
```

Comment:
I tried to cap off `MojiraBot.client` as far as possible. Would be good if it was not a necessary argument. Perhaps the `Channel` object should be passed over instead?

I mean it works like this, but I just try to avoid it in order to not give smaller classes power over the entire bot. That's not the point of modularization.

Reply:
IMO the process of getting a `Channel` from the `channel` config should be handled in the task instead of in the `MojiraBot` so that the main class wouldn't be a mess. Anyway I changed it to only parse channel. The changes can be rewinded at any time.